src/algebra/char_p/exp_char.lean

```diff
@@ -0,0 +1,138 @@
+/-
+Copyright (c) 2021 Jakob Scholbach. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Jakob Scholbach.
+-/
+import algebra.char_p.basic
+import algebra.char_zero
+import data.nat.prime
+
+/-!
+# Exponential characteristic
+
+This file defines the exponential characteristic of an integral domain, 
+and establishes a few basic facts relating the exponential characteristic to the (ordinary)
+characteristic.
+
+## Main results
+- `exp_char`: the definition of exponential characteristic
+- `exp_char_is_prime_or_one`: the exponential characteristic is a prime or one
+- `char_eq_exp_char_iff`: the characteristic equals the exponential characteristic iff the 
+  characteristic is prime
+
+## Tags
+exponential characteristic, characteristic
+-/
+
+universe u
+variables (R : Type u) [integral_domain R]
+
+/-- The definition of the exponential characteristic of a semiring. -/
+class exp_char (R : Type u) [semiring R] (q : ℕ) : Prop :=
+  (exp_char_def : (q = 1 ∧ char_zero R) ∨ (q.prime ∧ char_p R q))
```

Comment:
style: We don't indent lines after a `:=`
```suggestion
(exp_char_def : (q = 1 ∧ char_zero R) ∨ (q.prime ∧ char_p R q))
```

Reply:
Done (also similarly in other spots).